### PR TITLE
Add hikari-dev package to install development dependencies

### LINKED ISSUE #883: Add nox to dev-requirements
### Summary
It would make contributing to hikari a smoother experience if nox was actually installed as part of the dev reuqirements. It's a small change, but there's no reason it shouldn't be there (at least to my knowledge).

### Problem
People have to install nox separately. If they don't know what's going on when running nox, they may submit untested contributions.

### Ideal implementation
Something in the dev-requirements.txt. ~~Maybe also add a pytest patch in there too for being able to run tests on 3.11.~~

### Checklist
- [x] I have searched the issue tracker and have made sure it's not a duplicate.
      If it is a follow up of another issue, I have specified it.

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
Closes #883
Closes #886

### CODE REVIEW

**Review by FasterSpeeding — DISMISSED** [review]

**Comment by FasterSpeeding** [comment]
Tbh I don't really have much of an opinion on this either way

**Comment by FasterSpeeding** [comment]
Mind if i ask what the purpose or benefit of this approach over just having a couple of separate top level ({name}-)?requirement.txt files is?

File: hikari-dev/nox-requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+nox==2021.10.1
```
Comment (FasterSpeeding) [inline hikari-dev/nox-requirements.txt:1]: This nox ver is out of date

**Comment by davfsa** [comment]
> Mind if i ask what the purpose or benefit of this approach over just having a couple of separate top level ({name}-)?requirement.txt files is?

You are able to specify much gradually what you want to be installed. I have thought about having 1 file and then being able to parse it and then use it, but I havent had time to work on it.

I also thought about using setup.cfg, but decided against it when we talked about it the other day, as it might be deprecated and removed in the future.

#

I am still going to play a bit with this pr and try to get a nicer interface for it. Potentially have a script to which you specify the sections to install/remove and then have a specially formatted `dev-requirements.txt` which it parses. Will mark this pr as a draft until then. Hopefully can work on it in the upcoming week

**Comment by davfsa** [comment]
Superseded by #1238 (was easier to make a new pr that rebase this one)

**Pull request closed without merging**